src/entity/posts/CollectionPost.ts

```diff
@@ -0,0 +1,23 @@
+import { ChildEntity, Column } from 'typeorm';
+import { Post, PostType } from './Post';
+
+@ChildEntity(PostType.Collection)
+export class CollectionPost extends Post {
```

Comment:
Wouldn't it be better to inherit freeform?

Reply:
With @omBratteng DR in mind, perhaps we can create that subclass here? 
Or like you suggest inherit one that already shares all the fields :)

Reply:
Same argument as in that DR, when `FreeformPost` starts to get new fields then `CollectionPost` would get them as well which might not be what we want. For example we add `authorSignature` field to `FreeformPost`, now `CollectionPost` has it as well but that field does not make sense for that post type.

Reply:
I do believe we need some kind of composition or reusability here to keep it consistent. Any other suggestions?

Reply:
For sake of duplicate code I personally don't mind it. But agree it might be beneficial if we define a better relationship on which type of posts are similar and can extend from each other. The above with always extending from base `Post` is the simplest and allows us to move fast.

Reply:
My only alternative would be a sub type that's not actively used, but in return might as well use Post :)

Reply:
Yeah, are we good on merging this as is for now?

Reply:
@idoshamun is it ok from your end?

Reply:
yes